pandas/core/groupby/groupbyindexing.py

```diff
@@ -0,0 +1,214 @@
+from __future__ import annotations
+
+from pandas.util._decorators import doc
+import numpy as np
+
+
+class GroupByIndexingMixin:
+    """
+    Mixin for adding .iloc to GroupBy.
+    """
+
+    @property
+    def iloc(self) -> _ilocGroupByIndexer:
+        """
+        Integer location-based indexing for selection by position per group.
+
+        Similar to ``.apply(lambda x: x.iloc[i:j, k:l])``, but much faster and returns
+        a subset of rows from the original DataFrame with the original index and order
+        preserved.
+
+        The output is compatible with head() and tail()
+        The output is different from take() and nth() which do not preserve the index or order
+
+        Inputs
+        ------
+        Allowed inputs for the first index are:
+
+        - An integer, e.g. ``5``.
+        - A slice object with ints and positive step, e.g. ``1:``, ``4:-3:2``.
+
+        Allowed inputs for the second index are as for DataFrame.iloc, namely:
+
+        - An integer, e.g. ``5``.
+        - A list or array of integers, e.g. ``[4, 3, 0]``.
+        - A slice object with ints, e.g. ``1:7``.
+        - A boolean array.
+        - A ``callable`` function with one argument (the calling Series or
+          DataFrame) and that returns valid output for indexing (one of the above).
+
+        Returns
+        -------
+        Series or DataFrame
+
+        Note
+        ----
+        Neither GroupBy.nth() nor GroupBy.take() take a slice argument and
+        neither of them preserve the original DataFrame order and index.
+        They are both slow for large integer lists and take() is very slow for large group counts.
+
+        Use Case
```

Comment:
this is not a valid section head (you can instead put this in Notes & combine with Note). This won't render.